photongo/Readme.md

```diff
@@ -122,356 +154,484 @@ enum Role {
-)
+deleted, err := client.Post.DeleteMany.Where(
+  photon.Post.CreatedAt.Gt(christmas),
+).Exec(ctx)
 ```
 
-### Upsert
+#### Upsert
 
-#### Create a user or update their role
+##### Create a user or update their role
 
 ```go
-usr, err := client.User.Upsert(
-  user.New().Email("alice@prisma.io"),
-  user.New().Role(user.Role.ADMIN),
-  user.Where().Email("alice@prisma.io"),
-)
+user, err := client.User.Upsert.Where(
+  photon.User.Email.Equals("alice@prisma.io"),
+).Data(
+  photon.User.Email.Set("alice@prisma.io"),
+  photon.User.Role.Set(user.Role.ADMIN),
+).Exec(ctx)
 ```
 
-### Select
+#### Select multiple things
 
-#### Select a user, their posts and their comments
-
-We're using generated fields here to maintain type-safety
+##### Select a user with 10 of their posts
 
 ```go
-var u struct {
-  user.ID
-  user.Email
-  Posts []struct {
-    post.Title
-    post.CreatedAt
-    Comments []struct {
-      comment.Comment
-      comment.Text
-    }
-  }
-}
-
-err := client.User.Select(&u,
-  user.Where().ID("bobs-id"),
-  user.WithPosts(
-    post.Where().TitleContains("my title"),
-  ),
-)
+posts, err := client.User.FindOne.Where(
+  photon.User.Where.ID.Equals("bobs-id"),
+).With(
+  photon.User.Posts.
+    Where(
+      photon.Post.Title.Contains("some title")
+    ).
+    Limit(10),
+).Exec(ctx)
 ```
 
-### Raw
+##### Advanced selection of various nested relations
 
-You can use the raw fields in a model as an escape hatch to write custom complex queries.
+This will fetch users
+
+Note: This uses an additional relation "friends" which is not 
 
 ```go
-var u user.User
-err := client.User.Raw(&u, `select %s, %s from users`,
-  user.ID(10),
-  user.Email("alice@prisma.io")),
-)
+posts, err := client.User.FindOne.Where(
+  photon.User.Where.ID.Equals("bobs-id"),
+).With(
+  photon.User.Posts.
+    Where(
+      photon.Post.Title.Contains("some title")
+    ).
+    Limit(10),
+).Exec(ctx)
 ```
 
-## Adding Context
+### Raw usage
 
-Context can be added to a client with the following:
+#### Raw PQL Query
+
+The PQL function can be used as an escape hatch to query the Prisma server directly.
 
 ```go
-ctx := context.Background()
-client = client.WithContext(ctx)
+var u photon.User
+err := client.User.PQL(
+  ctx,
+  &u,
+  `
+    query {
+      findOnePost(where: {
+        id: "123"
+      }) {
+        id
+        title
+        content
+      }
+    }
+  `,
+  map[string]interface{}{
+    "id": "alice@prisma.io",
+  },
+)
 ```
 
-Normally you'd add this line at the beginning of your request handlers:
+#### Raw database query
+
+The raw function can be used as an escape hatch to write custom complex queries which are
+sent directly to the underlying database.
 
 ```go
-func (u *User) Create(w http.ResponseWriter, r *http.Request) {
-  client = client.WithContext(r.Context())
-  usr, err := client.User.Find(user.Where().ID(10))
-}
+var u photon.User
+err := client.User.Raw(
+  ctx,
+  &u,
+  `SELECT * FROM users WHERE email = ?`,
```

Comment:
Not a huge fan of the question marks since they're easy to mess up. Also MySQL specific. Have you seen this? 

https://github.com/prisma/specs/blob/raw-spec/photon-raw-api/Readme.md#photon-go

Do you like this syntax? Any adjustments you'd like to make?

Reply:
Fully agree here, thanks for the raw spec link. I just added a small example to have _something_.